File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, l1: Optional[ListNode], l2: Optional[ListNode]) -> Optional[ListNode]:
        a = l1 
        b = l2
        
        dummy = ListNode(0)
        
        c = dummy
        
        while a or b:
            if a is None:
                c.next = b
                break
            elif b is None:
                c.next = a
                break
                
            elif a.val <= b.val:
                c.next = a
                a = a.next
            else:
                c.next = b
                b = b.next
            c = c.next
        
        return dummy.next

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.java


class Solution {
    public ListNode mergeTwoLists(ListNode l1, ListNode l2){
             
        ListNode head;
        
        if( l1 == null) return l2; 
        if( l2 == null) return l1;
        
        if(l1.val < l2.val){
            head = l1;
        l1 = l1.next;
        } 
        
        else {
            head = l2;
            l2 = l2.next;
        }
        head.next = mergeTwoLists(l1,l2);
        return head;
    }
}